mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -41,6 +41,60 @@ CON_COMMAND_F(
     pPlayer->TogglePracticeMode();
 }
 
+CON_COMMAND(
+    mom_eyetele,
+    "Teleports the player to the solid that they are looking at.\n")
+{
+    CMomentumPlayer *pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer)
+        return;
+
+    if (pPlayer->m_bHasPracticeMode || !g_pMomentumTimer->IsRunning())
+    {
+        trace_t tr;
+        Vector pos = pPlayer->EyePosition();;
+        Vector ang;
+        pPlayer->EyeVectors(&ang);
+
+        int mask = CONTENTS_SOLID | CONTENTS_MOVEABLE | CONTENTS_OPAQUE | CONTENTS_WINDOW;
+        UTIL_TraceLine(pos, pos + ang * MAX_COORD_RANGE, mask, pPlayer, COLLISION_GROUP_NONE, &tr);
+
+        if (tr.fraction != 1.0 && tr.DidHit())
+        {
+            Vector hit = tr.endpos;
+            if (enginetrace->PointOutsideWorld(hit))
+            {
+                hit += (hit - pos).Normalized() * 64;
+
+                UTIL_TraceLine(hit, hit + ang * MAX_COORD_RANGE, mask, pPlayer, COLLISION_GROUP_NONE, &tr);
+
+                if (tr.DidHit())
```

Comment:
What happens if the new trace does not hit? It'll continue through this method -- is that intentional?

Reply:
This is if you use this from out of bounds, the trace will usually hit the outside of the level geometry, so it jumps forward a bit and does another trace so that you teleport to what you are actually aiming at and not the back of a wall.

Reply:
Okay I see. Worst case it just teleports to the place out of bounds.